### PR TITLE
Fix import_semantic_skill_from_directory

### LINKED ISSUE #235: import_semantic_skill_from_directory is not assigning config
**Describe the bug**
`import_semantic_skill_from_directory` is not reassigning the values with the result of the [config.from_json(config_file.read())](https://github.com/microsoft/semantic-kernel/blob/python-preview/python/semantic_kernel/kernel_extensions/import_semantic_skill_from_directory.py#L45) call resulting in an empty config.  

Easy fix is to just change line 45 to
```
config = config.from_json(config_file.read())
```

**To Reproduce**
1. Set a debugger [here](https://github.com/microsoft/semantic-kernel/blob/python-preview/python/semantic_kernel/kernel_extensions/import_semantic_skill_from_directory.py#L45) in the IDE. 
2. Run the following in Debug mode on IDE
```
import semantic_kernel as sk
from semantic_kernel.kernel_extensions.import_semantic_skill_from_directory import import_semantic_skill_from_directory

#create a kernel
kernel = sk.KernelBuilder.create_kernel()

#import skills
skills_directory = "../../skills" #path to skills directory
skill = import_semantic_skill_from_directory(kernel, skills_directory, "SummarizeSkill")
```
3. Look at the `config` object. Notice there's no description or default backends.

**Expected behavior**
`config` object has info from the json file such as `description` and `default_backends`.

**Screenshots**
image

**Desktop (please complete the following information):**
 - OS: Windows
 - IDE: PyCharm

### PR DESCRIPTION
**Pull request by renrut**
### Motivation and Context
This PR fixes #235 and adds a test for the functionality. This is ahead of the planning skill.

### Description
1. Fix to fix the typo in the `from_dict()` method in `prompt_template_config.py`
2. Add assignment back to skill config so results from JSON aren't discarded in `import_semantic_skill_from_directory.py`
3. Add tests

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [x] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: python/tests/test_config.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import os
```
Comment (dluc) [inline python/tests/test_config.py:1]: pls add usual header, copyright notice

**Comment by renrut** [comment]
So I also found an issue in the kernel.py [here](https://github.com/microsoft/semantic-kernel/blob/python-preview/python/semantic_kernel/kernel.py#L243) in the same code path. If the[ `prompt_template_config.from_dict()`](https://github.com/microsoft/semantic-kernel/blob/python-preview/python/semantic_kernel/semantic_functions/prompt_template_config.py#L55) doesn't parse out a `default_backend` from the config.json, it will result in a `NoneObject` and not an empty list and the kernel assumes empty list.

I can go ahead and fix it with a revision in this PR since it's fairly tangential to this change or open another?

Option 1: 
change kernel to check for None before trying to get length of the `default_backend`
```
function_config.prompt_template_config.default_backends if (function_config.prompt_template_config.default_backends and len(function_config.prompt_template_config.default_backends) > 0) else None
```

Option 2 (preferred):
change prompt_template_config from_dict parsing to default to empty list. I'd probably go with this option...
```
config.default_backends = data.get("default_backends", [])
```

**Comment by jjhenkel** [comment]
Thanks for all this @renrut, I agree w/ Option 2 and will take a look at this PR when that's in (looks great now, don't anticipate any changes).

This was a path I never tested in the original port (loading skills from a directory) so I really appreciate your help here! Thanks for the insights, the fixes, and generally looking through things and finding what's wrong!!

**Review by alexchaomander — APPROVED** [review]
Looks good to me! Thanks for adding these changes!

**Review by jjhenkel — APPROVED** [review]
Looks good to me, thanks again! (Sorry for the delay here, thanks for that additional fix!)

**Comment by renrut** [comment]
Sweet. I'm not authorized to merge, so I assume one of you all will.

**Review by dluc — APPROVED** [review]

**Pull request merged**